ext/sidecar.c

```diff
@@ -115,6 +116,10 @@ static void dd_sidecar_post_connect(ddog_SidecarTransport **transport, bool is_f
 }
 
 static void dd_sidecar_on_reconnect(ddog_SidecarTransport *transport) {
+    if (!ddtrace_endpoint || !dogstatsd_endpoint) {
+        return;
+    }
+
```

Comment:
The `dd_sidecar_connection_factory_ex` function uses another pattern:
https://github.com/DataDog/dd-trace-php/blob/6122feb8e17f47680f6269bb30af35b20936f251/ext/sidecar.c#L161-L165

I am not 100% certain why a `ZEND_ASSERT` is sufficient below, but most likely because `ddtrace_endpoint` and `dogstatsd_endpoint` should be kept in sync anyway.

So maybe this is just fine ..